plugins/base/src/main/kotlin/translators/documentables/DefaultPageCreator.kt

```diff
@@ -87,12 +91,24 @@ open class DefaultPageCreator(
 
     open fun pageForProperty(p: DProperty): MemberPageNode? = MemberPageNode(p.name, contentForProperty(p), setOf(p.dri), p)
 
+    private fun <T> T.isInherited(): Boolean where T: Documentable, T: WithExtraProperties<T> =
+        sourceSets.all { sourceSet -> extra[InheritedMember]?.isInherited(sourceSet) == true }
+
     private val WithScope.filteredFunctions: List<DFunction>
-        get() = functions.mapNotNull { function ->
-            function.takeIf {
-                it.sourceSets.any { sourceSet -> it.extra[InheritedFunction]?.isInherited(sourceSet) != true }
-            }
+        get() = functions.filterNot { it.isInherited() }
+
+    private val WithScope.filteredProperties: List<DProperty>
+        get() = properties.filterNot { it.isInherited() }
+
+    private fun <T> Collection<T>.split(): Pair<Set<T>, Set<T>> where T: Documentable, T: WithExtraProperties<T> {
```

Comment:
This method is not a generic split, please rename it to something more descriptive, like `splitByInherited`